samples/core/WPF/GetStartedWPF/GetStartedWPF/MainWindow.xaml.cs

```diff
@@ -0,0 +1,57 @@
+﻿using Microsoft.EntityFrameworkCore;
+using System.ComponentModel;
+using System.Windows;
+using System.Windows.Data;
+
+namespace GetStartedWPF
+{
+    /// <summary>
+    /// Interaction logic for MainWindow.xaml
+    /// </summary>
+    public partial class MainWindow : Window
+    {
+        private readonly ProductContext _context = 
+            new ProductContext();
+
+        private CollectionViewSource categoryViewSource =>
+            FindResource(nameof(categoryViewSource))
+            as CollectionViewSource;
```

Comment:
Could initialize this once in the constructor (after the InitializeComponent call) instead of finding it each time.

Reply:
But again, you might not even need it at all with INotifyPropertyChanged